src/lib/crypto.c

```diff
@@ -66,9 +68,34 @@ void crypto_extractPublicKey(uint32_t bip44Path[BIP44_LEN_DEFAULT], uint8_t *pub
     memcpy(pubKey, cx_publicKey.W, PK_LEN);
 }
 
+__Z_INLINE int blake_hash(const unsigned char *in, unsigned int inLen,
+               unsigned char *out, unsigned int outLen) {
+
+    cx_blake2b_t ctx;
+    cx_blake2b_init(&ctx, outLen * 8);
+    cx_hash(&ctx.header, CX_LAST, in, inLen, out, outLen);
+
+    return 0;
+}
+
+__Z_INLINE int blake_hash_cid(const unsigned char *in, unsigned int inLen,
+               unsigned char *out, unsigned int outLen) {
+
+    cx_blake2b_t ctx;
+    cx_blake2b_init(&ctx, outLen * 8);
+    cx_hash(&ctx.header, 0, prefix, inLen, NULL, 0);
```

Comment:
inLen is incorrect here 
```suggestion
    cx_hash(&ctx.header, 0, prefix, sizeof(prefix), NULL, 0);
```